examples/nlp/token_classification/punctuate_capitalize_infer.py

```diff
@@ -0,0 +1,739 @@
+        all_subtokens_mask: masks out all subwords besides the first one
+        all_quantities_of_preceding_words: number of words in query preceding a segment. Used for joining
+            predictions from overlapping segments.
+        all_query_ids: index of a query to which segment belongs
+        all_is_first: is segment first segment in a query
+        all_is_last: is segment last segment in a query
+    """
+    st = []
+    stm = []
+    sent_lengths = []
+    for i, query in enumerate(queries):
+        subtokens, subtokens_mask = _get_subtokens_and_subtokens_mask(query, tokenizer)
+        sent_lengths.append(len(subtokens))
+        st.append(subtokens)
+        stm.append(subtokens_mask)
+    _check_max_seq_length_and_margin_and_step(max_seq_length, margin, step)
+    max_seq_length = min(max_seq_length, max(sent_lengths) + 2)
+    logging.info(f'Max length: {max_seq_length}')
+    # Maximum number of word subtokens in segment. The first and the last tokens in segment are CLS and EOS
+    length = max_seq_length - 2
+    step = min(length - margin * 2, step)
+    get_stats(sent_lengths)
+    all_input_ids, all_segment_ids, all_subtokens_mask, all_input_mask, all_input_mask = [], [], [], [], []
+    all_quantities_of_preceding_words, all_query_ids, all_is_first, all_is_last = [], [], [], []
+    for q_i, query_st in enumerate(st):
+        q_inp_ids, q_segment_ids, q_subtokens_mask, q_inp_mask, q_quantities_of_preceding_words = [], [], [], [], []
+        for i in range(0, max(len(query_st), length) - length + step, step):
+            subtokens = [tokenizer.cls_token] + query_st[i : i + length] + [tokenizer.sep_token]
+            q_inp_ids.append(tokenizer.tokens_to_ids(subtokens))
+            q_segment_ids.append([0] * len(subtokens))
+            q_subtokens_mask.append([0] + stm[q_i][i : i + length] + [0])
+            q_inp_mask.append([1] * len(subtokens))
+            q_quantities_of_preceding_words.append(np.count_nonzero(stm[q_i][:i]))
+        all_input_ids.append(q_inp_ids)
+        all_segment_ids.append(q_segment_ids)
+        all_subtokens_mask.append(q_subtokens_mask)
+        all_input_mask.append(q_inp_mask)
+        all_quantities_of_preceding_words.append(q_quantities_of_preceding_words)
+        all_query_ids.append([q_i] * len(q_inp_ids))
+        all_is_first.append([True] + [False] * (len(q_inp_ids) - 1))
+        all_is_last.append([False] * (len(q_inp_ids) - 1) + [True])
+    return (
+        list(itertools.chain(*all_input_ids)),
+        list(itertools.chain(*all_segment_ids)),
+        list(itertools.chain(*all_input_mask)),
+        list(itertools.chain(*all_subtokens_mask)),
+        list(itertools.chain(*all_quantities_of_preceding_words)),
+        list(itertools.chain(*all_query_ids)),
+        list(itertools.chain(*all_is_first)),
+        list(itertools.chain(*all_is_last)),
+    )
+
+
+class BertPunctuationCapitalizationInferLongDataset(Dataset):
+    """
```

Comment:
@PeganovAnton @AlexGrinch let's move this to `https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/nlp/data/token_classification/punctuation_capitalization_dataset.py`. This inference script is way too long and most of the functions could be hidden